.github/workflows/open-source-unit-tests.yml

```diff
@@ -0,0 +1,136 @@
+name: Py-marqo  open source unit tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      push_to:
+        description: 'Docker registry location. Options: "ECR" or "DockerHub"'
+        required: true
+        default: 'DockerHub'
+      image_repo:
+        description: 'Marqo docker image repo name'
+        required: true
+        default: 'marqo'
+      image_tag:
+        description: 'Marqo image tag. Examples: "1.1.0", "test" "latest"'
+        required: true
+  push:
+    branches: 
+      - mainline
+  pull_request:
+    branches: 
+      - mainline
+
+permissions:
+  contents: read
+
+jobs:
+  Start-Runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_EC2_GH_RUNNER_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_EC2_GH_RUNNER_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id:  ${{ secrets.AMD_EC2_IMAGE_ID_200GB }}
```

Comment:
noting that this means we can't use this for any cuda-related py-marqo tests, but that should be fine, as we don't have any.

Reply:
Thanks for bringing this up Joshua, it made me consider this use case. 

I think this workflow will stick to CPU-only tests for now. These main set of unit tests should be runnable locally on developers' machines. We can always clone this workflow and change it to create CUDA-dedicated workflow (similar to what we do in the Marqo repo) in the future .